services/thermal_mgr/thermal_mgr.c

```diff
@@ -38,23 +42,47 @@ void initThermalSystemManager(lm75bd_config_t *config) {
   thermalMgrQueueHandle = xQueueCreateStatic(
     THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
     thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
+  if (event->type == OS_HANDLER_EVENT){
+    xQueueSendToFront(thermalMgrQueueHandle, event, ( TickType_t ) 0);
+  }  
+  else if (event->type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD){
+    xQueueSend(thermalMgrQueueHandle, event, ( TickType_t ) 10 );
+  }
```

Comment:
If someone adds a new event type in the future, this function won't send the event and will still return success. This would make it an annoying bug to solve. I'd either return an error code if an invalid event type is passed in or have a default block time for all other messages.